src/pages/auth/Signup.jsx

```diff
@@ -0,0 +1,80 @@
+import { Link } from "react-router-dom";
+import "./auth.css";
+import {useState} from "react";
+import { useAuth } from "../../context";
+
+export function Signup(){
+    const {signUpUser} = useAuth();
+    const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+    const [isConfirmPasswordVisible, setIsConfirmPasswordVisible] = useState(false);
+    const [credentials, setCredentials] = useState({
+        email: "",
+        firstName: "",
+        lastName: "",
+        password: ""
+    });
+    const [confirmPass, setConfirmPass] = useState("");
+    const [errMsg, setErrMsg] = useState("");
+
+    const passwordVisibilityHandler = (e) => setIsPasswordVisible(prev=> !prev);
+
+    const confirmPasswordVisibilityHandler = (e) => setIsConfirmPasswordVisible(prev=> !prev);
+    
+    const signUpHandler = () => {
+        if(!credentials.email || !credentials.password || !credentials.firstName || !credentials.lastName || !confirmPass){
+            setErrMsg("Kindly fill all the fields");
+        }else if(credentials.password !== confirmPass){
+            setErrMsg("Passwords donot match");
+        }else{
+            signUpUser(credentials);
+        }
+    }
+
+    return (
+        <div className="grid-container-auth">
+        <div className="form box-shadow">
+        <h3>Signup</h3>
+        {errMsg && <p className="err-msg">{errMsg}</p>}
+        <div className="input input-text">
+            <label>Email address</label>
+            <input type="email" value={credentials.email} placeholder="example@gmail.com" onChange={(e)=>setCredentials(prev=>({...prev, email: e.target.value}))} />
+        </div>
+        <div className="input input-text">
+            <label>First name</label>
+            <input type="text" value={credentials.firstName} placeholder="First name" onChange={(e)=>setCredentials(prev=>({...prev, firstName: e.target.value}))} />
+        </div>
+        <div className="input input-text">
+            <label>Last name</label>
+            <input type="text" value={credentials.lastName} placeholder="Last name" onChange={(e)=>setCredentials(prev=>({...prev, lastName: e.target.value}))} />
+        </div>
+        <div className="input input-text">
+            <label>Password</label>
+            <div className="password-wrapper">
+            <input type={isPasswordVisible?"text":"password"} value={credentials.password} placeholder="password" onChange={(e)=>setCredentials(prev=>({...prev, password: e.target.value}))} />
```

Comment:
Nitpick: You can extract callback for onChange

Reply:
okay will do it after submission